File: libs/document-processor-next/src/__tests__/models/set-property.ts
import { z } from 'zod';
import { withTypeBlueId } from '@blue-labs/language';

import { handlerContractBaseSchema } from '../../model/shared/index.js';

export const setPropertySchema = withTypeBlueId('SetProperty')(
  handlerContractBaseSchema.extend({
    propertyKey: z.string(),
    propertyValue: z.number().int(),
    path: z.string().optional(),
  })
);

export type SetProperty = z.infer<typeof setPropertySchema>;

File: references/java-sdk/src/test/java/blue/language/processor/model/SetProperty.java
package blue.language.processor.model;

import blue.language.model.TypeBlueId;
import blue.language.processor.model.HandlerContract;

@TypeBlueId("SetProperty")
public class SetProperty extends HandlerContract {

    private String propertyKey;
    private int propertyValue;
    private String path;

    public String getPropertyKey() {
        return propertyKey;
    }

    public void setPropertyKey(String propertyKey) {
        this.propertyKey = propertyKey;
    }

    public int getPropertyValue() {
        return propertyValue;
    }

    public void setPropertyValue(int propertyValue) {
        this.propertyValue = propertyValue;
    }

    public String getPath() {
        return path;
    }

    public void setPath(String path) {
        this.path = path;
    }
}
